yrs/src/block.rs

```diff
@@ -930,12 +934,15 @@ impl Item {
             }
             TypePtr::ID(id) => {
                 let ptr = store.blocks.get_block(id).unwrap();
-                let item = ptr.as_item().unwrap();
-                if let ItemContent::Type(branch) = &item.content {
-                    self.parent = TypePtr::Branch(BranchPtr::from(branch));
+                self.parent = if let Block::Item(item) = ptr.deref() {
+                    match &item.content {
+                        ItemContent::Type(branch) => TypePtr::Branch(BranchPtr::from(branch)),
+                        ItemContent::Deleted(_) => TypePtr::Unknown,
+                        _ => panic!("Defect: parent points to a block which is not a shared type"),
+                    }
                 } else {
-                    panic!("Defect: parent points to a block which is not a shared type");
```

Comment:
This panic could trigger i.e. if integrated item belonged to shared type that has been already removed.